### PR TITLE
Fix Bug 1666277: Do not use text-transform on localizable text

### PR DESCRIPTION
**Pull request by Mitch4sho**
Addressing bug 1666277: remove text-transform from associated localizable text and replace with uppercase in js and FTL files

### CODE REVIEW

File: frontend/src/core/editor/components/FailedChecks.js
Comment (abowler2) [inline frontend/src/core/editor/components/FailedChecks.js:128]: No need to transform this. While it is a button, it has its own className of `.close` which does not transform the text.

**Review by abowler2 — COMMENTED** [review]
Nice work!

I've noted a few things for you to address below. I did also notice that there appear to still be a number of instances of `text-transform: uppercase` remaining in the `/frontend` folder

File: frontend/src/modules/batchactions/components/BatchActions.js
Comment (Mitch4sho) [reply frontend/src/modules/batchactions/components/BatchActions.js:228]: Okay reverted this change

File: frontend/src/modules/batchactions/components/BatchActions.js
Comment (Mitch4sho) [reply frontend/src/modules/batchactions/components/BatchActions.js:239]: I believed I solved this, I dove inside these components and changed the text within them

File: frontend/public/static/locale/en-US/translate.ftl
Comment (mathjazz) [inline frontend/public/static/locale/en-US/translate.ftl:110]: I just noticed that the glyph (spinner) never appears when saving or approving translation. It seems unrelated to this patch, though. Can somebody confirm? We should file a bug.

**Review by mathjazz — COMMENTED** [review]
Nice work! Left a couple of comments. I'll let @abowler2 do the full review. :-)

Please also update `es` and `fr` FTL files.

File: frontend/public/static/locale/en-US/translate.ftl
Comment (Mitch4sho) [reply frontend/public/static/locale/en-US/translate.ftl:110]: Everything else is done I'm just going to wait on answer on this specific problem to see if I can do anything  about it right.

Comment (abowler2) [reply frontend/public/static/locale/en-US/translate.ftl:110]: @Mitch4sho I'll test this out when I have a chance to review this patch and update if there is something that needs/can be done in this patch or file a bug accordingly.

File: frontend/public/static/locale/es/translate.ftl
Comment (mathjazz) [inline frontend/public/static/locale/es/translate.ftl:12]: Mind the extra space.

File: frontend/public/static/locale/fr/translate.ftl
Comment (mathjazz) [inline frontend/public/static/locale/fr/translate.ftl:17]: I've noticed in several places across this file that the non-ASCII letters have changed now that they are uppercase.

For example here, I believe È should actually be É here.

And in other places ê became È instead of Ê.

File: frontend/public/static/locale/es/translate.ftl
Comment (Mitch4sho) [reply frontend/public/static/locale/es/translate.ftl:12]: Did you want me to remove the space so there are no spaces or just one space?

Comment (mathjazz) [reply frontend/public/static/locale/es/translate.ftl:12]: There's an extra ("COPIAR ") which should be removed ("COPIAR").

File: frontend/public/static/locale/fr/translate.ftl
Comment (adngdb) [inline frontend/public/static/locale/fr/translate.ftl:48]: "CLAVIER" instead of "CALVIER".

File: frontend/public/static/locale/fr/translate.ftl
Comment (adngdb) [inline frontend/public/static/locale/fr/translate.ftl:83]: "SAUVEGARDÉS" instead of "SAUVEGARDÈS".

**Review by adngdb — COMMENTED** [review]
Just quickly went through the French translation file. :relaxed:

File: frontend/public/static/locale/fr/translate.ftl
Comment (Mitch4sho) [reply frontend/public/static/locale/fr/translate.ftl:48]: Thank you!

File: frontend/public/static/locale/fr/translate.ftl
Comment (Mitch4sho) [reply frontend/public/static/locale/fr/translate.ftl:83]: Fixing this right away!

**Review by abowler2 — COMMENTED** [review]
Really nice progress. I've left a few comments below.

I also noticed a filter panel separator line that was missed [HERE](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/search/components/TimeRangeFilter.js#L306)

In addition there are still a few strings in need of transforming. Many of these are a bit trickier to find, so let me know if you would like some help with locating those.

File: frontend/public/static/locale/en-US/translate.ftl
Comment (Mitch4sho) [reply frontend/public/static/locale/en-US/translate.ftl:137]: got it!

File: frontend/public/static/locale/en-US/translate.ftl
Comment (Mitch4sho) [reply frontend/public/static/locale/en-US/translate.ftl:19]: Lord

File: frontend/src/core/editor/components/FailedChecks.js
Comment (Mitch4sho) [reply frontend/src/core/editor/components/FailedChecks.js:128]: Working on these right now

File: frontend/public/static/locale/en-US/translate.ftl
Comment (Pike) [reply frontend/public/static/locale/en-US/translate.ftl:137]: Should this be localizable at all? Sounds like a functional glyph rather than copy?

File: frontend/public/static/locale/en-US/translate.ftl
Comment (abowler2) [reply frontend/public/static/locale/en-US/translate.ftl:110]: @mathjazz The spinners should appear on this button as well as `Suggesting` and `Saving` but it does not appear that they are. What is interesting is that there are tests for both `Saving` and `Suggesting` that check for the class `.fa-spin` that are not failing. So, that leads me to believe that the class is being applied, but the spinner is not actually rendering. 

It looks like those were [added](https://github.com/mozilla/pontoon/pull/1564) a little while back, but are no longer working??  Would it be better to [reopen the original bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1614588) or file a new one?

Comment (mathjazz) [reply frontend/public/static/locale/en-US/translate.ftl:110]: Thanks. Reopened https://bugzilla.mozilla.org/show_bug.cgi?id=1614588 and added a comment with my findings.

**Comment by abowler2** [comment]
Great work!  The latest changes all look good!

There are just a couple more spots that need transforming:
1. In [Metadata](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/entitydetails/components/Metadata.css#L24): There are titles that will need to get transformed. Within Metadata each `render` function (ie. `renderComment`, `renderContext`) call the `Property` component and passes the 'title' as a prop. Each of those titles will need to be transformed. 

1. In [Machinery -> Translation](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/machinery/components/Translation.css#L15): Within the `<header>` there is a call to the [`TranslationSource` component](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/machinery/components/Translation.js#L77) which in turn calls to many different components (ie. `TranslationMemory`, `GoogleTranslation`, etc) which each render strings that will need to be transformed.

Those are the last 2 spots I found 😊

**Comment by abowler2** [comment]
@mathjazz I'm not sure [this selector](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/entitydetails/components/Metadata.css#L14) is being used any longer as I couldn't find any instances. Can you take a look and verify if it is indeed unused or if there are items that need transformed that fall under this selector.

**Comment by mathjazz** [comment]
> @mathjazz I'm not sure [this selector](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/entitydetails/components/Metadata.css#L14) is being used any longer as I couldn't find any instances. Can you take a look and verify if it is indeed unused or if there are items that need transformed that fall under this selector.

It's used here I think ("SINGULAR"):
https://pontoon.mozilla.org/sl/amo/all-resources/?string=66616

**Comment by Mitch4sho** [comment]
Alright working on it now!

**Comment by abowler2** [comment]
> It's used here I think ("SINGULAR"):
> https://pontoon.mozilla.org/sl/amo/all-resources/?string=66616

Yep, there it is 😃

@Mitch4sho These titles are rendered as part of the call to `OriginalStringProxy` component and can be found [here](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/entitydetails/components/GenericOriginalString.js#L36) and [here](https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/entitydetails/components/GenericOriginalString.js#L46).

File: frontend/public/static/locale/en-US/translate.ftl
Comment (abowler2) [inline frontend/public/static/locale/en-US/translate.ftl:229]: See more -> See More

File: frontend/src/modules/machinery/components/CaighdeanTranslation.js
Comment (abowler2) [inline frontend/src/modules/machinery/components/CaighdeanTranslation.js:19]: You won't need to transform the `title` here as in this case it is an attribute and not a prop. This is true for the other instances of `translation-source` as well.

File: frontend/src/modules/machinery/components/TranslationMemory.js
Comment (abowler2) [inline frontend/src/modules/machinery/components/TranslationMemory.js:36]: This title attribute also won't need transforming.

**Review by abowler2 — COMMENTED** [review]
Thanks for the update! Almost there 😄 

There are a few places where the transformation was made that won't need it, I've commented to point those out but let me know if you have questions.

Also, it looks like one of the tests broke and will need to be fixed.

File: frontend/src/modules/entitydetails/components/Metadata.js
Comment (abowler2) [inline frontend/src/modules/entitydetails/components/Metadata.js:313]: The failing test is looking for this element based in part on this title. So, you will need to update the test to reflect the uppercase status of this title in order for that test to pass.

**Review by abowler2 — APPROVED** [review]
Really nice job! Thanks so much for sticking it out on this 😊

**Comment by abowler2** [comment]
@mathjazz When you have a chance this is ready for your review.

File: frontend/public/static/locale/en-US/translate.ftl
Comment (mathjazz) [inline frontend/public/static/locale/en-US/translate.ftl:429]: SYSTRAN

**Review by mathjazz — COMMENTED** [review]
Very nice job, @Mitch4sho!

I've discovered two strings that are now lowercase, please uppercase them:
* `batchactions-BatchActions--header-selected-count`
* `history-Translation--toggle-diff`

Thank you for your great work making our codebase more localizable!

**Review by mathjazz — APPROVED** [review]
Patches like this one are particularly annoying and often drive people away. Thanks for staying persistent and finishing the job!

I'll merge #1727 first and I'll ask you to rebase if any conflicts pop up afterwards.

**Comment by Mitch4sho** [comment]
thanks, I'm glad I could help!

**Comment by mathjazz** [comment]
Please rebase and fix the conflicts.

**Comment by Mitch4sho** [comment]
Got it!

**Review by mathjazz — APPROVED** [review]
Thanks for the rebase!

**Comment by Mitch4sho** [comment]
It's finally merged!!

**Pull request merged**
